packages/desktop-client/src/hooks/useUndo.ts

```diff
@@ -0,0 +1,89 @@
+import { useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+
+import { undo, redo, addNotification } from 'loot-core/client/actions';
+import { type Notification } from 'loot-core/client/state-types/notifications';
+
+type UndoNotification = Pick<
+  Notification,
+  'type' | 'title' | 'message' | 'messageActions'
+>;
+
+type RedoNotification = Pick<
+  Notification,
+  'type' | 'title' | 'message' | 'messageActions'
+>;
+
+type UndoActions = {
+  undo: ReturnType<typeof undo>;
+  redo: ReturnType<typeof redo>;
+  showUndoNotification: (undoNotification: UndoNotification) => void;
+  showRedoNotification: (redoNotification: RedoNotification) => void;
+};
+
+export function useUndo(): UndoActions {
+  const dispatch = useDispatch();
+
+  const dispatchUndo = useCallback(async () => {
+    await dispatch(undo());
+  }, [dispatch]);
+
+  const dispatchRedo = useCallback(async () => {
+    await dispatch(redo());
+  }, [dispatch]);
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏the async/await here is unnecessary

```suggestion
  const dispatchUndo = useCallback(() => {
    dispatch(undo());
  }, [dispatch]);

  const dispatchRedo = useCallback(() => {
    dispatch(redo());
  }, [dispatch]);
```